File: Competitions/2023/Frozen Eagle/Earfquake/earfquake.java
import java.util.*;
import java.lang.Math;
class earfquake
{
    public static void main(String[] args)
    {
        int x;
        int y;
        int m;
        double total = 0;
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        for(;n>0;n--)
        {
            x = in.nextInt();
            y = in.nextInt();
            m = in.nextInt();
            double dist = Math.pow(x, 2) + Math.pow(y, 2);
            total += m * Math.pow(Math.E, -dist);
        }
        System.out.printf("%.3f\n", Math.abs(Math.log10(total)));
    }
}

File: Competitions/2023/Frozen Eagle/Earfquake/earfquake.py
import math

N = int(input())
t = 0

for _ in range(N):
    x, y, m = map(int, input().split())
    t += m * math.e**(-1*(x**2 + y**2))
    
print("%.3f" % -math.log10(t)) 
